cdap-app-templates/cdap-etl/cdap-etl-api/src/main/java/io/cdap/cdap/etl/api/MultiInputStageConfigurer.java

```diff
@@ -41,4 +43,21 @@
    * @param outputSchema output schema for this stage
    */
   void setOutputSchema(@Nullable Schema outputSchema);
+
+  /**
+   * Add validation failure to this configurer.
+   *
+   * @param message failure message
+   * @param correctiveAction corrective action
+   * @return a validation failure
+   */
+  ValidationFailure addFailure(String message, @Nullable String correctiveAction);
```

Comment:
why does this return the ValidationFailure?

Reply:
answered below